src/webgpu/shader/execution/expression/call/builtin/atomics/harness.ts

```diff
@@ -0,0 +1,61 @@
+import { GPUTest } from '../../../../../../gpu_test';
+
+export const workgroupSizes = [1, 2, 32, 64];
+export const dispatchSizes = [1, 4, 8, 16];
+
+export function runTest(
+  t: GPUTest,
+  workgroupSize: number, // Workgroup X-size
+  dispatchSize: number, // Dispatch X-size
+  bufferNumElements: number, // Number of 32-bit elements in output buffer
+  initValue: number, // 32-bit initial value used to fill output buffer
+  op: string, // Atomic op source executed by the compute shader
+  expected: Uint32Array // Expected values array to compare against output buffer
```

Comment:
nit: you get self-documenting arguments if you used a structure here. It also allows you to make some values optional, without imposing ordering constraints.